### PR TITLE
Check if device/app exists before removing it, when using application/device.remove()

### PR DESCRIPTION
**Pull request by mkorakakis**
Related CLI issues:
[#114](https://github.com/resin-io/resin-cli/issues/114)
[#123](https://github.com/resin-io/resin-cli/issues/123)

### CODE REVIEW

**Comment by jviotti** [comment]
Why does this PR has a description: `…/device.remove()`?

**Comment by jviotti** [comment]
Can you link the CLI issues that this PR is intended to fix?

**Comment by jviotti** [comment]
Can you add unit tests for this as well? You can stub `pine.get` to return an empty array, which will make `remove()` fail. Testing only the case when the resource doesn't exist should be enough.

**Comment by mkorakakis** [comment]
So, something like this?
```
		describe '.remove()', ->

			describe 'given an invalid application', ->

				beforeEach ->
					@pineGetStub = m.sinon.stub(pine, 'get')
					@pineGetStub.returns(Promise.resolve([]))

				afterEach ->
					@pineGetStub.restore()

				it 'should reject with not found error', ->
					promise = application.remove('MyApp')
					m.chai.expect(promise).to.be.rejectedWith(errors.ResinApplicationNotFound)
```

**Comment by jviotti** [comment]
@mkorakakis Looks good!

File: tests/models/device.spec.coffee [hunk 1]
```diff
@@ -487,6 +487,21 @@ describe 'Device Model:', ->
...
+			describe 'given the device does not exist', ->
+
+				beforeEach ->
+					@pineGetStub = m.sinon.stub(pine, 'get')
+					@pineGetStub.returns(Promise.resolve([]))
+
+				afterEach ->
+					@pineGetStub.restore()
+
+				it 'should be rejected', ->
```
Comment (jviotti) [inline tests/models/device.spec.coffee]: Can you add `with not found error` as the other unit test?

**Pull request merged**